### PR TITLE
Request user confirmation before removing logo

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
Request user confirmation before removing logo

## Motivation and Context
JTBD from Branding epic.

## How Has This Been Tested?
Manually tested removing a logo and cancel/confirm clicks.
Also confirm the modal doesn't show for regular images.

https://apps-stage-1.risevision.com

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

**Note**: The button color does not match mockup due to a recent styling change that introduced this defect, also present in Presentation Delete modal. Reported here: https://github.com/Rise-Vision/common-header/pull/1039#discussion_r316344775

It will get automatically fixed once styling is reverted. We are also performing QA to confirm this is not shipped to users. 

@alex-deaconu Please review. Thanks!

### CODE REVIEW

File: web/scripts/template-editor/components/directives/dtv-component-image.js [hunk 1]
```diff
@@ -262,7 +262,9 @@ angular.module('risevision.template-editor.directives')
           };
 
           $scope.removeImageFromList = function (image) {
-            _setSelectedImages(imageFactory.removeImage(image, $scope.selectedImages));
+            imageFactory.canRemoveImage(image).then(function(){
+              _setSelectedImages(imageFactory.removeImage(image, $scope.selectedImages));
```
Comment (alex-deaconu) [inline web/scripts/template-editor/components/directives/dtv-component-image.js]: @ezequielc `imageFactory.canRemoveImages()` be part of `imageFactory.removeImage()`? ie. not add another factory function. `removeImage()` could just return the old images list if the user wants to keep the image. 

Also, does `removeImage` require the `image` parameter? I don't see it being used.

Thoughts? Thanks.

Comment (ezequielc) [reply web/scripts/template-editor/components/directives/dtv-component-image.js]: @alex-deaconu I was struggling to have a good solution given the asynchronous call but I like your suggestion. I will update the code.
And I agree, there is no need to pass `image` as a parameter. Thanks!

File: web/scripts/template-editor/components/services/svc-base-image-factory.js [hunk 2]
```diff
@@ -31,9 +31,7 @@ angular.module('risevision.template-editor.services')
         var metadata =
           fileMetadataUtilsService.metadataWithFileRemoved(currentMetadata, image);
 
-        if (metadata) {
```
Comment (alex-deaconu) [inline web/scripts/template-editor/components/services/svc-base-image-factory.js]: @ezequielc do we not need this check anymore? Thanks.

Comment (ezequielc) [reply web/scripts/template-editor/components/services/svc-base-image-factory.js]: @alex-deaconu It doesn't look like this case can happen as the user will click Remove in a file from the metadata list.

I would expect the `metadataWithFileRemoved` to return the original metadata if file is not found, but it looks like there is no return value in such case. 
To be safe I will add a conditional and resolve to an empty metadata. I don't want to change the behaviour of the other method as it is being used elsewhere. Do you agree?

Comment (alex-deaconu) [reply web/scripts/template-editor/components/services/svc-base-image-factory.js]: @ezequielc agreed, thanks!

Comment (ezequielc) [reply web/scripts/template-editor/components/services/svc-base-image-factory.js]: @alex-deaconu Updated, please review. Thanks!

**Pull request merged**